### PR TITLE
feat(form): add datepicker widget

### PR DESCRIPTION
**Pull request by henrikbossart**
## What does this pull request change?
Adds a date widget to the form, instead of using the Stringfield.

### CODE REVIEW

File: example/app/data/DemoDataSource/recipes/plugins/form/simple/simple.recipe.json [hunk 1]
```diff
@@ -12,8 +12,14 @@
...
-          "type": "PLUGINS:dm-core-plugins/form/fields/StringField",
-          "widget": "DateTimeWidget"
+          "type": "PLUGINS:dm-core-plugins/form/fields/DateField",
+          "widget": "DateTimeWidget",
+          "config": {
+            "type": "PLUGINS:dm-core-plugins/form/fields/DateFieldConfig",
+            "variant": "datetime",
+            "label": "Date & Time",
+            "useMinutes": true
+          }
```
Comment (soofstad) [inline example/app/data/DemoDataSource/recipes/plugins/form/simple/simple.recipe.json:22]: All of these are optional right?
So setting the type to "DateField" gives you the default config

Comment (soofstad) [inline example/app/data/DemoDataSource/recipes/plugins/form/simple/simple.recipe.json:22]: All of these are optional right?
So setting the type to "DateField" gives you the default config

**Review by soofstad — APPROVED** [review]

File: example/app/data/DemoDataSource/recipes/plugins/form/simple/simple.recipe.json (hunk 1 above)
Comment (henrikbossart) [reply example/app/data/DemoDataSource/recipes/plugins/form/simple/simple.recipe.json:22]: correct

**Review by eoaksnes — APPROVED** [review]
💯 just delete the [DateField.json](https://github.com/equinor/dm-core-packages/pull/909/files#diff-80c7bafee1a993b6716bd57d1be8ecc87fafa319c31155552224e91c64490036)

**Pull request merged**